components/channel_members_rhs/channel_members_rhs.tsx

```diff
@@ -50,46 +49,92 @@ export interface Props {
         closeRightHandSide: () => void;
         goBack: () => void;
         setChannelMembersRhsSearchTerm: (terms: string) => void;
-        loadProfilesAndReloadChannelMembers: (page: number, perParge: number, channelId: string) => void;
+        loadProfilesAndReloadChannelMembers: (page: number, perParge: number, channelId: string, sort: string) => void;
         loadMyChannelMemberAndRole: (channelId: string) => void;
         setEditChannelMembers: (active: boolean) => void;
         searchProfilesAndChannelMembers: (term: string, options: any) => Promise<{data: UserProfile[]}>;
     };
 }
 
+export interface ListItem {
+    type: 'member' | 'first-separator' | 'separator';
+    data: ChannelMember | JSX.Element;
+}
+
 export default function ChannelMembersRHS({
     channel,
     searchTerms,
     membersCount,
     canGoBack,
     teamUrl,
-    channelAdmins,
     channelMembers,
     canManageMembers,
     editing = false,
     actions,
 }: Props) {
+    const [list, setList] = useState<ListItem[]>([]);
+
+    const [page, setPage] = useState(0);
+    const [isNextPageLoading, setIsNextPageLoading] = useState(false);
+
     const searching = searchTerms !== '';
 
     // show search if there's more than 20 or if the user have an active search.
     const showSearch = searching || membersCount >= 20;
 
-    let normalMemberTitle;
-    if (channelMembers.length > 0 && !searching) {
-        normalMemberTitle = (
-            <FormattedMessage
-                id='channel_members_rhs.list.channel_members_title'
-                defaultMessage='MEMBERS'
-            />
-        );
-    }
-
     useEffect(() => {
         return () => {
             actions.setChannelMembersRhsSearchTerm('');
         };
     }, []);
 
+    useEffect(() => {
+        const listcp: ListItem[] = [];
+        let memberDone = false;
+
+        for (let i = 0; i < channelMembers.length; i++) {
+            const member = channelMembers[i];
+            if (listcp.length === 0) {
+                let text = null;
+                if (member.membership?.scheme_admin === true) {
+                    text = (
+                        <FormattedMessage
+                            id='channel_members_rhs.list.channel_admin_title'
+                            defaultMessage='CHANNEL ADMINS'
+                        />
+                    );
+                } else {
+                    text = (
+                        <FormattedMessage
+                            id='channel_members_rhs.list.channel_members_title'
+                            defaultMessage='MEMBERS'
+                        />
+                    );
+                    memberDone = true;
+                }
+
+                listcp.push({
+                    type: 'first-separator',
```

Comment:
nit: constants for types?

Reply:
Absolutely I'm gonna make a mew type. Thanks for the suggestion